### DISCUSSION TITLE
can't use x-for object in x-spread directive

### DISCUSSION THREAD

**Discussion by JonathanFSilva** [alpinejs/alpine#1090] [1. Help]
Hey guys!

I'm trying to use x-for object in x-spread directive.

```html
<table x-data="table()">
  <thead>
    <tr>
      <template x-for="column in columns">
        <th x-spread="column"></th>
      </template>
    </tr>
  </thead>
  <tbody>
    <template x-for="item in data">
      <tr>
        <template x-for="col in columns">
          <td x-text="item[col.key]"></td>
        </template>
      </tr>
    </template>
  </tbody>
</table>
```

```javascript
function table() {
  return {
    columns: [
      {
        ['key']: 'firstName',
        ['@click']: ($event) => doSomenthing($event),
        ['x-bind:class']: () => 'table__cell',
        ['x-text']: () => 'First Name',
      },
      {
        ['key']: 'lastName',
        ['@click']: ($event) => doSomenthing($event),
        ['x-bind:class']: () => 'table__cell',
        ['x-text']: () => 'Last Name',
      },
      {
        ['key']: 'email',
        ['@click']: ($event) => doSomenthing($event),
        ['x-bind:class']: () => 'table__cell',
        ['x-text']: () => 'Email',
      },
    ],
    data: [
      { firstName: 'John', lastName: 'Doe', email: 'john.doe@mail.com' },
    ],
  };
}
```

But i'm getting this error:

> Alpine Error: "ReferenceError: column is not defined"
Expression: "column"
Element: <th x-spread=​"column">​</th>​

Can someone please help me to understand why this code is not working?

Demo: https://codepen.io/jonathanfsilva/pen/OJbWVbN

**Comment by ryangjchandler** [comment]
@JonathanFSilva Not sure if this would be considered a bug or not to be honest. `x-spread` isn't evaluated in the context of the `x-for`, meaning any variables used inside of the loop won't be available. Instead, `x-spread` is evaluated in the context of the entire component.

Without knowing how the internal works, I would actually expect your code above to work. Unfortuantely that isn't the case - perhaps @calebporzio could weigh in on a reason why this wasn't supported and if it is going to be supported in the future.
